### ISSUE TITLE
Suggest to drop/skip an intermediate stop/waypoint if driver passed it or drives a different way

### ISSUE THREAD

**Issue by biodranik** [organicmaps/organicmaps#9528]
> Typically, I like to plan my route beforehand by adding waypoints (intermediate stops) along my desired route. This is fine until I encounter something unexpected that forces me to deviate from my planned route and miss one of my waypoints. The app will keep trying to make me turn around and go back to the missed WP. Is there any way to, in effect, tell OM to skip the next WP and recalculate the route from my current position?

For example, a dialog with something like "Skip [intermediate stop name]?" and "yes"/"no" may appear, with the default timer set to either yes or no buttons (TBD).

**Comment by 0709wiwiwi** [comment]
An example by another app.
Cruiser uses both **not announced** Shaping Points and **annouced** Via Points. 
In the menu you set (for both points) **or** strict (must visit) **or not** (skipable).
Strict points can still be skipped this by a manual action as described next.
https://github.com/devemux86/cruiser/discussions/212#discussioncomment-6153046

**Comment by dremelts** [comment]
Another example is OsmAnd.  You can place a "Quick Action" button on the screen which pops up a customizable list of single-tap actions you can perform.  One of the options that you can put on the list is Delete Nearest Destination.

**Comment by ZeljkoBG** [comment]
Another example is Automap. When adding a VIA point, it has the option to choose whether it is a **point** or **area**. If you continue or skip the VIA area, the route continues. VIA area is ignored.

**Comment by hornetster** [comment]
Totally agree with this request - can almost make the route useless.
I ride a motorcycle, and there needs to be a simple, minimal distraction method os skipping a waypoint.
This is particularly annoying, when planning a route, and if you select a town, it tries to take you through central park there!!
If it was possible to set an algorithm  when selecting a town, to either select a point on the road you are travelling, or at least, zoom in and allow you to select a point when creating route?
Thanks.

**Comment by Rob-from-VI** [comment]
Agreed, I have also requested this kind of thing once previously.   It is disconcerting to be asked to make a u-turn when it isn't necessary. 

However, this can be complicated by the situation where I mis-drive, but still want to hit the waypoint at some time before proceeding too much further.   

I certainly like the idea of a pop-up button that would allow me to re-confirm or cancel my waypoint if it appears by some parameters that I have passed or missed it.

I also like the idea of being able to set Shaping Points as opposed to Via points when devising my route.   A shaping point would be automatically deleted once one gets near it.  Now we need to define near???
